toolchain/check/handle_binding_pattern.cpp

```diff
@@ -219,18 +219,19 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
         context.AddNameToLookup(name_id, bind_id);
         auto entity_name_id =
             context.insts().GetAs<SemIR::AnyBindName>(bind_id).entity_name_id;
+        CARBON_CHECK(context.bind_name_cache()
+                         .Insert(entity_name_id, bind_id)
+                         .is_inserted());
```

Comment:
IIRC we've said in the past that we don't want side-effects in `CHECK` conditions because, while they're always evaluated now, we might revisit that choice in the future. (As a reader I also tend to skip over `CHECK`s when working out what the code does.)
```suggestion
        bool inserted = context.bind_name_cache()
                         .Insert(entity_name_id, bind_id)
                         .is_inserted();
        CARBON_CHECK(inserted);
```